doc/source/cluster/kubernetes/user-guides/persist-kuberay-operator-logs.md

```diff
@@ -0,0 +1,110 @@
+
+* Tail log files from Kubernetes containers.
+* Parse multi-line logs for Docker and CRI (Container Runtime Interface) formats.
+* Enrich logs with Kubernetes metadata such as namespace, pod, and container names.
+* Send the logs to Loki for centralized storage and querying.
+```{literalinclude} ../configs/loki.log.yaml
+:language: yaml
+:start-after: Fluent Bit Config
+:end-before: ---
+```
+
+A few notes on the above config:
+
+* Inputs: The `tail` input reads log files from `/var/log/containers/*.log`, with `multiline.parser` to handle complex log messages across multiple lines.
+* Filters: The `kubernetes` filter adds metadata like namespace, pod, and container names to each log, enabling more efficient log management and querying in Loki.
+* Outputs: The `loki` output block specifies Loki as the target. The `Host` and `Port` define the Loki service endpoint, and `Labels` add metadata for easier querying in Grafana. Additionally, `tenant_id` allows for multi-tenancy if required by the Loki setup.
+
+Deploy the Fluent Bit deployment with the [Helm chart repository](https://github.com/fluent/helm-charts/tree/main/charts/fluent-bit).
+
+```shell
+helm repo add fluent https://fluent.github.io/helm-charts
+helm repo update
+
+helm install fluent-bit fluent/fluent-bit -f fluent-bit-config.yaml
+```
+
+### Install the KubeRay Operator and Deploy a RayCluster
+
+Follow [this document](kuberay-operator-deploy) to install the KubeRay operator and deploy a RayCluster.
+
+### Deploy Grafana
+
+Create a `datasource-config.yaml` file with the following configuration to set up Grafana's Loki datasource:
+```{literalinclude} ../configs/loki.log.yaml
+:language: yaml
+:start-after: Grafana Datasource Config
+```
+
+Deploy the Grafana deployment with the [Helm chart repository](https://github.com/grafana/helm-charts/tree/main/charts/grafana).
+
+```shell
+helm repo add grafana https://grafana.github.io/helm-charts
+helm repo update
+
+helm install grafana grafana/grafana -f datasource-config.yaml
+```
+
+### Check the Grafana Dashboard
+
+Get the admin user password by running the following command:
+
+```shell
+kubectl get secret --namespace default grafana -o jsonpath="{.data.admin-password}" | base64 --decode ; echo
+```
+
+
+```shell
+# Verify that the Grafana pod is running in the `default` namespace.
+kubectl get pods --namespace default -l "app.kubernetes.io/name=grafana"
+# NAME                       READY   STATUS    RESTARTS   AGE
+# grafana-54d5d747fd-5fldc   1/1     Running   0          8m21s
+```
+
+To access Grafana from your local machine, set up port forwarding by running:
+```shell
+export POD_NAME=$(kubectl get pods --namespace default -l "app.kubernetes.io/name=grafana,app.kubernetes.io/instance=grafana" -o jsonpath="{.items[0].metadata.name}")
+kubectl --namespace default port-forward $POD_NAME 3000
+```
+
+This will make Grafana available locally at `http://localhost:3000`.
```

Comment:
This will make Grafana available locally at `http://localhost:3000`.
* Username: "admin"
* Password: Get the password using the following command:
  ```sh
  kubectl get secret --namespace default grafana -o jsonpath="{.data.admin-password}" | base64 --decode ; echo
  ```